packages/mgt/src/graph/graph.user.ts

```diff
@@ -111,20 +111,23 @@ export async function getMe(graph: IGraph): Promise<User> {
  * @memberof Graph
  */
 export async function getUserWithPhoto(graph: IGraph, userId?: string): Promise<IDynamicPerson> {
-  let person = null as IDynamicPerson;
-  let cache: CacheStore<CacheUser>;
   let photo = null;
   let user: IDynamicPerson;
+
   let cachedPhoto: CachePhoto;
+  let cachedUser: CacheUser;
+
   const resource = userId ? `users/${userId}` : 'me';
   const scopes = userId ? ['user.readbasic.all'] : ['user.read'];
 
   // attempt to get user and photo from cache if enabled
   if (usersCacheEnabled()) {
-    cache = CacheService.getCache<CacheUser>(cacheSchema, userStore);
-    const cachedUser = await cache.getValue(userId || 'me');
+    let cache = CacheService.getCache<CacheUser>(cacheSchema, userStore);
+    cachedUser = await cache.getValue(userId || 'me');
```

Comment:
I think there is something up with the way we cache non-person entities. I keep seeing the `cacheduer.user` return `undefined` for non-person people entities (like conference rooms). This throws an error because `JSON.parse` doesn't like `undefined` values. 
![image](https://user-images.githubusercontent.com/25832310/95500423-a6338400-095b-11eb-8a5f-2bb723742e68.png)
![image](https://user-images.githubusercontent.com/25832310/95500439-a9c70b00-095b-11eb-86e8-07e817cd8f87.png)

Reply:
Good catch - added a check before attempting to parse